libs/nx-terraform/src/generators/create-environment-sp/generator.ts

```diff
@@ -0,0 +1,97 @@
+import { Tree } from '@nrwl/devkit'
+import execa from 'execa'
+import { isDryRun } from '../../common/isDryRun'
+import { readRepoSettings } from '../../common/read-repo-settings'
+import { readConfigFromEnvFile } from '../../common/readConfigFromEnvFile'
+import { NxTerraformCreateEnvironmentServicePrincipalSchema } from './schema'
+
+export default async function (
+    tree: Tree,
+    options: NxTerraformCreateEnvironmentServicePrincipalSchema,
+) {
+    const repoSettings = readRepoSettings()
+    const environmentConfig = await readConfigFromEnvFile(
+        repoSettings.terraformStateType,
+        options.environmentName,
+    )
+
+    const servicePrincipalName = `gh-actions-${repoSettings.azureResourcePrefix}-${options.environmentName}-sp`
+    const scopes = `/subscriptions/${environmentConfig.subscriptionId}/resourcegroups/${environmentConfig.resourceGroupName}`
+    const containerScope = `/subscriptions/${environmentConfig.subscriptionId}/resourceGroups/${environmentConfig.resourceGroupName}/providers/Microsoft.Storage/storageAccounts/${environmentConfig.terraformStorageAccount}/blobServices/default/containers/${environmentConfig.terraformStorageContainer}`
+
+    if (isDryRun()) {
+        console.log('Will run:')
+
+        console.log(`> az ad sp create-for-rbac --role ${options.role} \
+    --scopes ${scopes} \
+    --name ${servicePrincipalName} \
+    --sdk-auth`)
+
+        console.log(`> az role assignment create \
+    --role "Storage Blob Data Contributor" \
```

Comment:
Just realised this is simply echoing what's gonna be run. So my comment above about skip-assignment is more for below when you actually call this command.

Another thought though - this dry run docco that gets consoled could get out of sync with what's actually run if the command array is changed but not the docco. Maybe too high-effort but would be cool if the docco of what's about to be run could be driven directly by the command array.

Reply:
Fixed this, the idea was to give you a command you could manually send to the admins if you didn't have permissions